code/ATMOSPHERICS/atmospherics.dm

```diff
@@ -65,6 +65,8 @@ GLOBAL_DATUM_INIT(pipe_icon_manager, /datum/pipe_icon_manager, new())
 // Icons/overlays/underlays
 /obj/machinery/atmospherics/update_icon()
 	var/turf/T = loc
+	if(!istype(T)) // Somehow loc became null
+		qdel(src)

```

Comment:
What if loc is, say, a locker or other container?
It can be things other than null, even if it isn't a turf.

Reply:
What you want is `var/turf/T = get_turf(src)`, and skip istype check.
`istype` is the wrong fix here anyway. If you don't like `loc` getting null, you should investigate why didn't it qdel its contents upon becoming `null`, and not just paper over with an istype check. Deleting `src` in `update_icon` of all places is.. weird

Reply:
It might be due to what kyet posted. That the thing is in a container. I'll change it

Reply:
Edit. I got no clue why that wouldn't be a turf or null. I'll change it to be more robust to runtimes but won't find the source of the issue